ocw/lib/EC2.py

```diff
@@ -334,3 +337,48 @@ def cleanup_images(self, valid_period_days):
                     else:
                         self.log_dbg("Delete image '{}' (ami:{})".format(img['Name'], img['ImageId']))
                         self.ec2_client(region).deregister_image(ImageId=img['ImageId'], DryRun=False)
+
+    def cleanup_k8s_jobs(self):
+        self.create_credentials_file()
+        subprocess.call(["aws", "sts", "get-caller-identity"])
+
+        clusters = {}
+        for region in self.cluster_regions:
+            response = self.eks_client(region).list_clusters()
+            if 'clusters' in response and len(response['clusters']) > 0:
+                clusters[region] = []
+                for cluster_name in response['clusters']:
+                    subprocess.call(["rm", "-r", "~/.kube"])
+                    subprocess.call(["aws", "eks", "update-kubeconfig", "--region", region, "--name", cluster_name])
+
+                    config.load_kube_config()
+                    k8s_client = client.BatchV1Api()
+                    now = datetime.now(timezone.utc)
+                    if self.dry_run:
+                        dry_run = "All"
+                    else:
+                        dry_run = None
+
+                    ret = k8s_client.list_job_for_all_namespaces(watch=False)
+                    for i in ret.items:
+                        age = (now - i.status.start_time).days
+                        if age > 1:
+                            self.log_info(f"Deleting from {cluster_name} the job {i.metadata.name} with age {age}")
+                            k8s_client.delete_namespaced_job(i.metadata.name, i.metadata.namespace, dry_run=dry_run)
+
+    def create_credentials_file(self):
+        access_key = self.getData("access_key")
```

Comment:
we already have this variables in private values

Reply:
done